pkg/virtual/initializingworkspaces/builder/forwarding.go

```diff
@@ -100,3 +106,134 @@ func provideForwardingRestStorage(ctx context.Context, clusterClient dynamic.Clu
+			statusSpec = &apiextensions.CustomResourceSubresourceStatus{}
+		}
+
+		var scaleSpec *apiextensions.CustomResourceSubresourceScale
+
+		strategy := customresource.NewStrategy(
+			typer,
+			namespaceScoped,
+			kind,
+			schemaValidator,
+			statusSchemaValidate,
+			map[string]*structuralschema.Structural{resource.Version: structuralSchema},
+			statusSpec,
+			scaleSpec,
+		)
+
+		storage, statusStorage := registry.NewStorage(
+			ctx,
+			resource,
+			"", // ClusterWorkspaces have no identity
+			kind,
+			listKind,
+			strategy,
+			nil,
+			tableConvertor,
+			nil,
+			clusterClient,
+			nil,
+			withUpdateValidation(initializer),
+		)
+
+		// we want to expose some but not all the allowed endpoints, so filter by exposing just the funcs we need
+		subresourceStorages = make(map[string]rest.Storage)
+		if statusEnabled {
+			subresourceStorages["status"] = &struct {
+				registry.FactoryFunc
+				registry.DestroyerFunc
+
+				registry.GetterFunc
+				registry.UpdaterFunc
+				// patch is implicit as we have get + update
+
+				registry.TableConvertorFunc
+				registry.CategoriesProviderFunc
+				registry.ResetFieldsStrategyFunc
+			}{
+				FactoryFunc:   statusStorage.FactoryFunc,
+				DestroyerFunc: statusStorage.DestroyerFunc,
+
+				GetterFunc:  statusStorage.GetterFunc,
+				UpdaterFunc: statusStorage.UpdaterFunc,
+
+				TableConvertorFunc:      statusStorage.TableConvertorFunc,
+				CategoriesProviderFunc:  statusStorage.CategoriesProviderFunc,
+				ResetFieldsStrategyFunc: statusStorage.ResetFieldsStrategyFunc,
+			}
+		}
+
+		// only expose GET
+		return &struct {
+			registry.FactoryFunc
+			registry.ListFactoryFunc
+			registry.DestroyerFunc
+
+			registry.GetterFunc
+
+			registry.TableConvertorFunc
+			registry.CategoriesProviderFunc
+			registry.ResetFieldsStrategyFunc
+		}{
+			FactoryFunc:     storage.FactoryFunc,
+			ListFactoryFunc: storage.ListFactoryFunc,
+			DestroyerFunc:   storage.DestroyerFunc,
+
+			GetterFunc: storage.GetterFunc,
+
+			TableConvertorFunc:      storage.TableConvertorFunc,
+			CategoriesProviderFunc:  storage.CategoriesProviderFunc,
+			ResetFieldsStrategyFunc: storage.ResetFieldsStrategyFunc,
+		}, subresourceStorages
+	}
+}
+
+// withUpdateValidation adds further validation to ensure that a user of this virtual workspace can only
+// remove their own initializer from the list
+func withUpdateValidation(initializer tenancyv1alpha1.ClusterWorkspaceInitializer) registry.StorageWrapper {
+	return func(resource schema.GroupResource, storage *registry.StoreFuncs) *registry.StoreFuncs {
+		delegateUpdater := storage.UpdaterFunc
+		storage.UpdaterFunc = func(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *v1.UpdateOptions) (runtime.Object, bool, error) {
+			validation := rest.ValidateObjectUpdateFunc(func(ctx context.Context, obj, old runtime.Object) error {
```

Comment:
Given the way we construct the server, only updates to status subresource are possible (no updater is exposed for spec). We could also go further and say that you can *only* update `status.initializers`, and then only to remove yours, but IDK if the value is there.

Reply:
I want to have controllers update ClusterWorkspaceStatus with conditions, so perhaps this is enough.